src/TypeSpec.Extension/Emitter.Csharp/src/lib/clientModelBuilder.ts

```diff
@@ -197,58 +192,38 @@ export function createModelForService(
 
     function addChildClients(
         context: EmitContext<NetEmitterOptions>,
-        client: SdkClient | DotnetSdkOperationGroup,
+        client: SdkClient | SdkOperationGroup,
         clients: InputClient[]
     ) {
-        if (emitterOptions.branded) {
-            const dpgOperationGroups = listOperationGroups(
-                sdkContext,
-                client as SdkClient
-            );
-            for (const dpgGroup of dpgOperationGroups) {
-                var dotnetOperationGroup = {
-                    ...dpgGroup,
-                    name: dpgGroup.type.name
-                };
-                var subClient = emitClient(dotnetOperationGroup, client);
-                clients.push(subClient);
-                addChildClients(context, dotnetOperationGroup, clients);
-            }
-        } else {
-            const dpgOperationGroups = listOperationGroupsByClient(
-                context,
-                client
-            );
-            for (const dpgGroup of dpgOperationGroups) {
-                clients.push(emitClient(dpgGroup, client));
-                addChildClients(context, dpgGroup, clients);
-            }
+        const dpgOperationGroups = listOperationGroups(
+            sdkContext,
+            client as SdkClient
+        );
+        for (const dpgGroup of dpgOperationGroups) {
+            var subClient = emitClient(dpgGroup, client);
+            clients.push(subClient);
+            addChildClients(context, dpgGroup, clients);
         }
     }
 
-    function getClientName(
-        client: SdkClient | DotnetSdkOperationGroup
-    ): string {
-        if (emitterOptions.branded) {
-            return client.kind === ClientKind.SdkClient
-                ? client.name
-                : client.type.name;
-        } else {
-            return client.kind === ClientKind.SdkClient
-                ? `${client.name}Client`
-                : client.name === "Models"
-                ? "ModelsOps"
-                : client.name; //quick fix for reserved namespace need something more robust
+    function getClientName(client: SdkClient | SdkOperationGroup): string {
+        if (client.kind === ClientKind.SdkClient) {
+            return client.name;
         }
+
+        var pathParts = client.groupPath.split(".");
+        return pathParts?.length >= 3
+            ? pathParts.slice(pathParts.length - 2).join("")
+            : client.type.name === "Models"
```

Comment:
Full tests will be in https://github.com/Azure/autorest.csharp/pull/4082

Reply:
I have concern to add this hard-code here which will make our customer confusing.

Reply:
Yes, we would discuss and handle that in #4082